src/main/java/org/traccar/protocol/EgtsProtocolDecoder.java

```diff
@@ -121,19 +127,29 @@ protected Object decode(
 
         ByteBuf buf = (ByteBuf) msg;
 
+        List<Position> positions = new LinkedList<>();
+
+        short headerLength = buf.getUnsignedByte(buf.readerIndex() + 3);
+        int frameDataLength = buf.getUnsignedShortLE(buf.readerIndex() + 5);
         int index = buf.getUnsignedShort(buf.readerIndex() + 5 + 2);
-        buf.skipBytes(buf.getUnsignedByte(buf.readerIndex() + 3));
+        short packetType = buf.getUnsignedByte(buf.readerIndex() + 5 + 2 + 2);
+        buf.skipBytes(headerLength);
 
-        List<Position> positions = new LinkedList<>();
+        if (packetType == PT_RESPONSE) {
+            // some retranslator systems do send response on response packets - just skip them
+            return null;
+        }
 
-        while (buf.readableBytes() > 2) {
+        int frameDataEnd = buf.readerIndex() + frameDataLength;
```

Comment:
Why do we need this change?

Reply:
I see it also was usable with unsplitted frames, reverted